src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -139,6 +146,8 @@ export class ProgrammingExerciseUpdateComponent implements OnInit {
         if (this.programmingExercise.id !== undefined) {
             this.problemStatementLoaded = true;
         }
+        // Checks if the current environment is production
+        this.inProductionEnvironment = this.profileService.isInProduction();
```

Comment:
please use a subscription and an async way here, e.g like the following code. 

        this.profileService.getProfileInfo().subscribe((profileInfo) => {
            if (profileInfo && profileInfo.guidedTourMapping) {
                this.guidedTourMapping = profileInfo.guidedTourMapping;
            }
        });

Reply:
!

Reply:
i added the a subscription for the profileInfo